events-application-rest-api/src/test/java/com/bounswe/bounswe2017group3/Model/UserControllerTest.java

```diff
@@ -0,0 +1,77 @@
+package com.bounswe.bounswe2017group3.Model;
+import com.bounswe.bounswe2017group3.Application;
+import com.bounswe.bounswe2017group3.Model.User;
+import com.bounswe.bounswe2017group3.Controller.UserController;
+import com.bounswe.bounswe2017group3.Repository.UserRepository;
+import com.jayway.restassured.RestAssured;
+import static com.jayway.restassured.RestAssured.when;
+import com.jayway.restassured.module.mockmvc.RestAssuredMockMvc;
+import static com.jayway.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import java.util.Arrays;
+import org.apache.http.HttpStatus;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = Application.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class UserControllerTest {
+ 
+    @Autowired   
+    UserRepository repository;
+    User usr1;
+    User usr2;
+    User usr3;
+    @Value("${local.server.port}")   
+    int port;
+ 
+    @Before
+    public void setUp() {
+        
+        usr1 = new User("gokce","Gökçe Uludoğan", "gokce@gmail.com","123123");
+        usr2 = new User("samed","Samed Düzçay","smddzcy@gmail.com","123456");
+        usr3 = new User("esra","Esra Aydemir","esra-aydemir@gmail.com","222333");        
+        repository.deleteAll();
```

Comment:
You are right :smile: I didn't realize that it deletes our actual repository :smile: So, I have changed it with the one using Mocks